src/Stream/FileWriter.cpp

```diff
@@ -40,7 +40,12 @@ namespace Stream
 	{
 		// Create directory if it does not exist. ofstream will fail if directory does not exist.
 		auto directory = XFile::GetDirectory(filename);
-		if (!directory.empty() && !XFile::PathExists(directory)) {
+		if (!directory.empty() && !XFile::PathExists(directory)) 
+		{
+			if (openMode != OpenMode::CanOpenNew) {
```

Comment:
This should test for the specific flag with `&`, rather than the combination of all flags with exact equality/inequality.
```
# Short form
if (openMode & OpenMode::CanOpenNew) {
```
```
# Explicit form
if ((openMode & OpenMode::CanOpenNew) != 0) {
```
```
# Overly explicit form
if ((openMode & OpenMode::CanOpenNew) == OpenMode::CanOpenNew) {
```

I think I would prefer the first form.

Reply:
I think I might also suggest testing the `OpenMode` flag before checking for a non-empty directory component to the path, or checking the filesystem itself.

Reply:
Thanks for catching my error on the flag check. Should be fixed with newest push.

If we check the OpenMode flag first, if a new directory is required, the code would bypass our custom error message and attempt to call ofstream. This would produce an error, but not as descriptive. Maybe I'm missing your point though...

-Brett

Reply:
Ahh, I didn't notice the custom error message.